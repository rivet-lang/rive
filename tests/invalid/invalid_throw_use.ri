struct ThrowError < Throwable {
    pub func to_string(&self) -> string {
        _ = self;
        return "ThrowError instance";
    }
}

func throw_with_never() -> never {
    throw ThrowError();
}

func throw_inside_defer() -> ! {
    defer {
        throw ThrowError();
    }
}

test "invalid `throw` inside `test` declaration" {
    throw ThrowError();
}

func invalid_throwing_function() {
    throw ThrowError();
}

func invalid_throwing_value() -> !int32 {
    throw 5;
}
