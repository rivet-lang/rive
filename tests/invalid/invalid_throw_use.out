lib/core/src/wyhash.ri:7:2: error: invalid preprocessing directive: `[unsafe`
    7 | #[unsafe; inline]
      |  ^~~~~~~~
lib/core/src/wyhash.ri:12:2: error: invalid preprocessing directive: `[inline`
   12 | #[inline]
      |  ^~~~~~~~
lib/core/src/wyhash.ri:17:2: error: invalid preprocessing directive: `[inline`
   17 | #[inline]
      |  ^~~~~~~~
lib/core/src/wyhash.ri:22:2: error: invalid preprocessing directive: `[inline`
   22 | #[inline]
      |  ^~~~~~~~
lib/core/src/lib.ri:23:2: error: invalid preprocessing directive: `[inline`
   23 | #[inline]
      |  ^~~~~~~~
lib/core/src/lib.ri:32:2: error: invalid preprocessing directive: `[inline`
   32 | #[inline]
      |  ^~~~~~~~
lib/core/src/lib.ri:41:2: error: invalid preprocessing directive: `[inline`
   41 | #[inline]
      |  ^~~~~~~~
lib/core/src/lib.ri:50:2: error: invalid preprocessing directive: `[inline`
   50 | #[inline]
      |  ^~~~~~~~
lib/core/src/float.ri:10:6: error: invalid preprocessing directive: `[inline`
   10 |     #[inline]
      |      ^~~~~~~~
lib/core/src/float.ri:15:6: error: invalid preprocessing directive: `[inline`
   15 |     #[inline]
      |      ^~~~~~~~
lib/core/src/float.ri:25:6: error: invalid preprocessing directive: `[inline`
   25 |     #[inline]
      |      ^~~~~~~~
lib/core/src/float.ri:32:6: error: invalid preprocessing directive: `[inline`
   32 |     #[inline]
      |      ^~~~~~~~
lib/core/src/float.ri:54:6: error: invalid preprocessing directive: `[inline`
   54 |     #[inline]
      |      ^~~~~~~~
lib/core/src/float.ri:61:6: error: invalid preprocessing directive: `[inline`
   61 |     #[inline]
      |      ^~~~~~~~
lib/core/src/StringBuilder.ri:7:2: error: invalid preprocessing directive: `[boxed`
    7 | #[boxed]
      |  ^~~~~~~
lib/core/src/StringBuilder.ri:12:6: error: invalid preprocessing directive: `[inline`
   12 |     #[inline]
      |      ^~~~~~~~
lib/core/src/StringBuilder.ri:17:6: error: invalid preprocessing directive: `[inline`
   17 |     #[inline]
      |      ^~~~~~~~
lib/core/src/StringBuilder.ri:28:6: error: invalid preprocessing directive: `[inline`
   28 |     #[inline]
      |      ^~~~~~~~
lib/core/src/StringBuilder.ri:33:6: error: invalid preprocessing directive: `[unsafe`
   33 |     #[unsafe; inline]
      |      ^~~~~~~~
lib/core/src/StringBuilder.ri:40:6: error: invalid preprocessing directive: `[unsafe`
   40 |     #[unsafe]
      |      ^~~~~~~~
lib/core/src/StringBuilder.ri:50:6: error: invalid preprocessing directive: `[inline`
   50 |     #[inline]
      |      ^~~~~~~~
lib/core/src/StringBuilder.ri:79:6: error: invalid preprocessing directive: `[inline`
   79 |     #[inline]
      |      ^~~~~~~~
lib/core/src/StringBuilder.ri:90:6: error: invalid preprocessing directive: `[inline`
   90 |     #[inline]
      |      ^~~~~~~~
lib/core/src/StringBuilder.ri:95:6: error: invalid preprocessing directive: `[inline`
   95 |     #[inline]
      |      ^~~~~~~~
lib/core/src/StringBuilder.ri:100:6: error: invalid preprocessing directive: `[inline`
  100 |     #[inline]
      |      ^~~~~~~~
lib/core/src/StringBuilder.ri:105:6: error: invalid preprocessing directive: `[inline`
  105 |     #[inline]
      |      ^~~~~~~~
lib/core/src/StringBuilder.ri:110:6: error: invalid preprocessing directive: `[inline`
  110 |     #[inline]
      |      ^~~~~~~~
lib/core/src/StringBuilder.ri:115:6: error: invalid preprocessing directive: `[inline`
  115 |     #[inline]
      |      ^~~~~~~~
lib/core/src/int.ri:11:6: error: invalid preprocessing directive: `[inline`
   11 |     #[inline]
      |      ^~~~~~~~
lib/core/src/int.ri:22:6: error: invalid preprocessing directive: `[inline`
   22 |     #[inline]
      |      ^~~~~~~~
lib/core/src/int.ri:33:6: error: invalid preprocessing directive: `[inline`
   33 |     #[inline]
      |      ^~~~~~~~
lib/core/src/int.ri:91:6: error: invalid preprocessing directive: `[inline`
   91 |     #[inline]
      |      ^~~~~~~~
lib/core/src/int.ri:155:6: error: invalid preprocessing directive: `[inline`
  155 |     #[inline]
      |      ^~~~~~~~
lib/core/src/int.ri:161:6: error: invalid preprocessing directive: `[inline`
  161 |     #[inline]
      |      ^~~~~~~~
lib/core/src/uint.ri:9:6: error: invalid preprocessing directive: `[inline`
    9 |     #[inline]
      |      ^~~~~~~~
lib/core/src/uint.ri:17:6: error: invalid preprocessing directive: `[inline`
   17 |     #[inline]
      |      ^~~~~~~~
lib/core/src/uint.ri:24:6: error: invalid preprocessing directive: `[inline`
   24 |     #[inline]
      |      ^~~~~~~~
lib/core/src/uint.ri:30:6: error: invalid preprocessing directive: `[inline`
   30 |     #[inline]
      |      ^~~~~~~~
lib/core/src/uint.ri:36:6: error: invalid preprocessing directive: `[inline`
   36 |     #[inline]
      |      ^~~~~~~~
lib/core/src/uint.ri:43:6: error: invalid preprocessing directive: `[inline`
   43 |     #[inline]
      |      ^~~~~~~~
lib/core/src/uint.ri:51:6: error: invalid preprocessing directive: `[inline`
   51 |     #[inline]
      |      ^~~~~~~~
lib/core/src/uint.ri:57:6: error: invalid preprocessing directive: `[inline`
   57 |     #[inline]
      |      ^~~~~~~~
lib/core/src/uint.ri:63:6: error: invalid preprocessing directive: `[inline`
   63 |     #[inline]
      |      ^~~~~~~~
lib/core/src/uint.ri:69:6: error: invalid preprocessing directive: `[inline`
   69 |     #[inline]
      |      ^~~~~~~~
lib/core/src/uint.ri:75:6: error: invalid preprocessing directive: `[inline`
   75 |     #[inline]
      |      ^~~~~~~~
lib/core/src/uint.ri:81:6: error: invalid preprocessing directive: `[inline`
   81 |     #[inline]
      |      ^~~~~~~~
lib/core/src/uint.ri:87:6: error: invalid preprocessing directive: `[inline`
   87 |     #[inline]
      |      ^~~~~~~~
lib/core/src/uint.ri:93:6: error: invalid preprocessing directive: `[inline`
   93 |     #[inline]
      |      ^~~~~~~~
lib/core/src/uint.ri:103:6: error: invalid preprocessing directive: `[inline`
  103 |     #[inline]
      |      ^~~~~~~~
lib/core/src/uint.ri:113:6: error: invalid preprocessing directive: `[inline`
  113 |     #[inline]
      |      ^~~~~~~~
lib/core/src/uint.ri:202:6: error: invalid preprocessing directive: `[inline`
  202 |     #[inline]
      |      ^~~~~~~~
lib/core/src/uint.ri:208:6: error: invalid preprocessing directive: `[inline`
  208 |     #[inline]
      |      ^~~~~~~~
lib/core/src/string.ri:9:2: error: invalid preprocessing directive: `[boxed`
    9 | #[boxed]
      |  ^~~~~~~
lib/core/src/string.ri:15:6: error: invalid preprocessing directive: `[unsafe`
   15 |     #[unsafe]
      |      ^~~~~~~~
lib/core/src/string.ri:25:6: error: invalid preprocessing directive: `[unsafe`
   25 |     #[unsafe]
      |      ^~~~~~~~
lib/core/src/string.ri:70:6: error: invalid preprocessing directive: `[inline`
   70 |     #[inline]
      |      ^~~~~~~~
lib/core/src/string.ri:174:6: error: invalid preprocessing directive: `[inline`
  174 |     #[inline]
      |      ^~~~~~~~
lib/core/src/string.ri:407:6: error: invalid preprocessing directive: `[inline`
  407 |     #[inline]
      |      ^~~~~~~~
lib/core/src/string.ri:458:6: error: invalid preprocessing directive: `[inline`
  458 |     #[inline]
      |      ^~~~~~~~
lib/core/src/string.ri:490:6: error: invalid preprocessing directive: `[inline`
  490 |     #[inline]
      |      ^~~~~~~~
lib/core/src/string.ri:636:6: error: invalid preprocessing directive: `[inline`
  636 |     #[inline]
      |      ^~~~~~~~
lib/core/src/string.ri:654:6: error: invalid preprocessing directive: `[inline`
  654 |     #[inline]
      |      ^~~~~~~~
lib/core/src/string.ri:676:6: error: invalid preprocessing directive: `[inline`
  676 |     #[inline]
      |      ^~~~~~~~
lib/core/src/string.ri:681:6: error: invalid preprocessing directive: `[inline`
  681 |     #[inline]
      |      ^~~~~~~~
lib/core/src/string.ri:686:6: error: invalid preprocessing directive: `[inline`
  686 |     #[inline]
      |      ^~~~~~~~
lib/core/src/string.ri:692:6: error: invalid preprocessing directive: `[inline`
  692 |     #[inline]
      |      ^~~~~~~~
lib/core/src/string.ri:726:6: error: invalid preprocessing directive: `[inline`
  726 |     #[inline]
      |      ^~~~~~~~
lib/core/src/string.ri:731:6: error: invalid preprocessing directive: `[inline`
  731 |     #[inline]
      |      ^~~~~~~~
lib/core/src/string.ri:761:6: error: invalid preprocessing directive: `[inline`
  761 |     #[inline]
      |      ^~~~~~~~
lib/core/src/string.ri:778:6: error: invalid preprocessing directive: `[inline`
  778 |     #[inline]
      |      ^~~~~~~~
lib/core/src/string.ri:783:6: error: invalid preprocessing directive: `[inline`
  783 |     #[inline]
      |      ^~~~~~~~
lib/core/src/mem.ri:49:2: error: invalid preprocessing directive: `[inline`
   49 | #[inline]
      |  ^~~~~~~~
lib/core/src/mem.ri:60:2: error: invalid preprocessing directive: `[inline`
   60 | #[inline]
      |  ^~~~~~~~
lib/core/src/mem.ri:73:2: error: invalid preprocessing directive: `[inline`
   73 | #[inline]
      |  ^~~~~~~~
lib/core/src/mem.ri:107:2: error: invalid preprocessing directive: `[inline`
  107 | #[inline]
      |  ^~~~~~~~
lib/core/src/mem.ri:113:2: error: invalid preprocessing directive: `[unsafe`
  113 | #[unsafe; inline]
      |  ^~~~~~~~
lib/core/src/array.ri:13:2: error: invalid preprocessing directive: `[inline`
   13 | #[inline]
      |  ^~~~~~~~
lib/core/src/array.ri:18:2: error: invalid preprocessing directive: `[inline`
   18 | #[inline]
      |  ^~~~~~~~
lib/core/src/array.ri:35:2: error: invalid preprocessing directive: `[inline`
   35 | #[inline]
      |  ^~~~~~~~
lib/core/src/bool.ri:6:6: error: invalid preprocessing directive: `[inline`
    6 |     #[inline]
      |      ^~~~~~~~
lib/core/src/errors.ri:17:2: error: invalid preprocessing directive: `[inline`
   17 | #[inline]
      |  ^~~~~~~~
lib/core/src/errors.ri:24:2: error: invalid preprocessing directive: `[boxed`
   24 | #[boxed]
      |  ^~~~~~~
lib/core/src/errors.ri:28:6: error: invalid preprocessing directive: `[inline`
   28 |     #[inline]
      |      ^~~~~~~~
lib/core/src/errors.ri:34:2: error: invalid preprocessing directive: `[boxed`
   34 | #[boxed]
      |  ^~~~~~~
lib/core/src/errors.ri:38:6: error: invalid preprocessing directive: `[inline`
   38 |     #[inline]
      |      ^~~~~~~~
lib/core/src/errors.ri:44:2: error: invalid preprocessing directive: `[boxed`
   44 | #[boxed]
      |  ^~~~~~~
lib/core/src/errors.ri:48:6: error: invalid preprocessing directive: `[inline`
   48 |     #[inline]
      |      ^~~~~~~~
lib/core/src/errors.ri:65:6: error: invalid preprocessing directive: `[inline`
   65 |     #[inline]
      |      ^~~~~~~~
lib/core/src/Vector.ri:5:2: error: invalid preprocessing directive: `[boxed`
    5 | #[boxed]
      |  ^~~~~~~
lib/core/src/Vector.ri:13:6: error: invalid preprocessing directive: `[unsafe`
   13 |     #[unsafe; inline]
      |      ^~~~~~~~
lib/core/src/Vector.ri:18:6: error: invalid preprocessing directive: `[unsafe`
   18 |     #[unsafe; inline]
      |      ^~~~~~~~
lib/core/src/Vector.ri:23:6: error: invalid preprocessing directive: `[unsafe`
   23 |     #[unsafe]
      |      ^~~~~~~~
lib/core/src/Vector.ri:30:6: error: invalid preprocessing directive: `[unsafe`
   30 |     #[unsafe; inline]
      |      ^~~~~~~~
lib/core/src/Vector.ri:35:6: error: invalid preprocessing directive: `[unsafe`
   35 |     #[unsafe; inline]
      |      ^~~~~~~~
lib/core/src/Vector.ri:125:6: error: invalid preprocessing directive: `[inline`
  125 |     #[inline]
      |      ^~~~~~~~
lib/core/src/Vector.ri:137:6: error: invalid preprocessing directive: `[inline`
  137 |     #[inline]
      |      ^~~~~~~~
lib/core/src/Vector.ri:173:6: error: invalid preprocessing directive: `[inline`
  173 |     #[inline]
      |      ^~~~~~~~
lib/core/src/process.ri:41:2: error: invalid preprocessing directive: `[inline`
   41 | #[inline]
      |  ^~~~~~~~
lib/core/src/StaticBuffer.ri:34:6: error: invalid preprocessing directive: `[inline`
   34 |     #[inline]
      |      ^~~~~~~~
lib/core/src/console.ri:17:2: error: invalid preprocessing directive: `[inline`
   17 | #[inline]
      |  ^~~~~~~~
lib/core/src/console.ri:48:2: error: invalid preprocessing directive: `[inline`
   48 | #[inline]
      |  ^~~~~~~~
lib/core/src/console.ri:53:2: error: invalid preprocessing directive: `[unsafe`
   53 | #[unsafe]
      |  ^~~~~~~~
lib/core/src/console.ri:66:2: error: invalid preprocessing directive: `[unsafe`
   66 | #[unsafe]
      |  ^~~~~~~~
lib/core/src/backtrace.ri:7:2: error: invalid preprocessing directive: `!`
    7 | #![compile_c_source("../thirdparty/libbacktrace/backtrace.c")]
      |  ^~
lib/core/src/rune.ri:58:6: error: invalid preprocessing directive: `[inline`
   58 |     #[inline]
      |      ^~~~~~~~
lib/core/src/rune.ri:104:2: error: invalid preprocessing directive: `[inline`
  104 | #[inline]
      |  ^~~~~~~~
lib/core/src/rune.ri:117:2: error: invalid preprocessing directive: `[boxed`
  117 | #[boxed]
      |  ^~~~~~~
rivet: error: could not compile module `invalid_throw_use`, aborting due to 107 previous errors